### PR TITLE
Idf scorer

### PR DESCRIPTION
**Pull request by rosequ**
- IDF overlapping without considering TF
- added custom stop word list
- considered document score as tie breaker

### CODE REVIEW

File: src/main/resources/io/anserini/qa/english-stoplist.txt [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+of
+off
+on
+once
+only
+or
+other
+ought
+our
+ours	ourselves
```
Comment (lintool) [inline src/main/resources/io/anserini/qa/english-stoplist.txt]: typo here?

File: src/main/resources/io/anserini/qa/english-stoplist.txt [hunk 2]
```diff
@@ -0,0 +1,174 @@
+a
```
Comment (lintool) [inline src/main/resources/io/anserini/qa/english-stoplist.txt]: where does this file come from?

File: src/main/java/io/anserini/qa/passage/WmdPassageScorer.java [hunk 3]
```diff
@@ -0,0 +1,7 @@
+package io.anserini.qa.passage;
```
Comment (lintool) [inline src/main/java/io/anserini/qa/passage/WmdPassageScorer.java]: remove empty file from PR?

Comment (rosequ) [inline src/main/java/io/anserini/qa/passage/WmdPassageScorer.java]: Oh, sorry, I thought I had it removed!!!

File: src/main/java/io/anserini/qa/passage/PassageScorer.java [hunk 4]
```diff
@@ -16,10 +16,11 @@
 package io.anserini.qa.passage;
 
 import java.util.List;
+import java.util.Map;
 
 public interface PassageScorer {
 
-  void score(List<String> sentences, String output) throws Exception;
+  void score(String question, Map<String, Float> sentences) throws Exception;
```
Comment (lintool) [inline src/main/java/io/anserini/qa/passage/PassageScorer.java]: would it make sense to make the input a sorted map?

File: src/main/resources/io/anserini/qa/english-stoplist.txt (hunk 2 above)
Comment (rosequ) [inline src/main/resources/io/anserini/qa/english-stoplist.txt]: I used one of those available online. I think it's from here: http://www.ranks.nl/stopwords

File: src/main/java/io/anserini/qa/passage/PassageScorer.java (hunk 4 above)
Comment (rosequ) [inline src/main/java/io/anserini/qa/passage/PassageScorer.java]: Of course.. But, I will have to go through the entire input, anyway.

File: src/main/java/io/anserini/qa/passage/WmdPassageScorer.java (hunk 3 above)
Comment (lintool) [inline src/main/java/io/anserini/qa/passage/WmdPassageScorer.java]: Why not use a more standard list, e.g., one from NLTK?
http://www.nltk.org/book/ch02.html

File: src/main/java/io/anserini/py4j/PyseriniEntryPoint.java [hunk 5]
```diff
@@ -26,11 +26,9 @@
 import io.anserini.rerank.RerankerCascade;
 import org.apache.lucene.search.similarities.BM25Similarity;
 import org.apache.lucene.search.similarities.Similarity;
-import java.util.List;
-import java.util.ArrayList;
-import java.util.TreeMap;
-import java.util.Map;
-import java.util.SortedMap;
+
+import java.util.*;
```
Comment (lintool) [inline src/main/java/io/anserini/py4j/PyseriniEntryPoint.java]: Let's do explicit imports?
See http://stackoverflow.com/questions/3348816/intellij-never-use-wildcard-imports

Comment (rosequ) [inline src/main/java/io/anserini/py4j/PyseriniEntryPoint.java]: Sorry. This again is Intellij's doing

File: src/main/java/io/anserini/qa/passage/IdfPassageScorer.java [hunk 6]
```diff
@@ -38,33 +41,42 @@
...
   private final int topPassages;
+  private final List<String> stopWords;
 
   public IdfPassageScorer(String index, Integer k) throws IOException {
     this.util = new IndexUtils(index);
     this.directory = FSDirectory.open(new File(index).toPath());
     this.reader = DirectoryReader.open(directory);
     this.topPassages = k;
     scoredPassageHeap = MinMaxPriorityQueue.maximumSize(topPassages).create();
+    stopWords = IOUtils.readLines(new FileInputStream("src/main/resources/io/anserini/qa/english-stoplist.txt"));
```
Comment (lintool) [inline src/main/java/io/anserini/qa/passage/IdfPassageScorer.java]: Use getResource here. See: http://stackoverflow.com/questions/2593154/get-a-resource-using-getresource

With getResource, when we distribute Anserini as a jar, if it's packaged correctly, the stopwords list will still be available.

**Pull request merged**